[2025-05-20 15:29:06] [INFO] [common.py:76] <module>: Initializing the common module.
[2025-05-20 15:29:06] [INFO] [project.py:35] <module>: Loading libclang from "C:/Program Files/LLVM/lib/libclang.lib".
[2025-05-20 15:29:06] [ERROR] [project.py:47] <module>: Failed to load libclang with the error: LibclangError('[WinError 193] %1 is not a valid Win32 application. To provide a path to libclang use Config.set_library_path() or Config.set_library_file().')
[2025-05-20 15:29:07] [INFO] [project.py:131] get_project_list_from_config: Loading the project "Glibc" (glibc) with the following configurations: {'short_name': 'glibc', 'database_id': 5, 'database_name': 'glibc', 'github_data_name': 'glibc', 'vendor_id': [72], 'product_id': [767], 'url_pattern': 'sourceware', 'master_branch': 'master', 'language': 'c', 'include_directory_path': 'include', 'repository_path': 'C:/Users/ramya/Documents/university/2 ano/Database project/Glibc/glibc', 'output_directory_path': 'C:/Users/ramya/Documents/university/2 ano/Database project/CVEDetailsScrape/Output', 'scrape_all_branches': False, 'output_directory_diff_path': 'C:/Users/ramya/Documents/university/2 ano/Database project/CVEDetailsScrape/Output/diff_between_files'}
[2025-05-20 15:29:07] [INFO] [project.py:92] __init__: Loaded the project "Glibc" located in "C:/Users/ramya/Documents/university/2 ano/Database project/Glibc/glibc".
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Apache HTTP Server" (apache).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Xen" (xen).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Linux Kernel" (kernel).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Mozilla" (mozilla).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "MongoDB" (mongodb).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "MySQL" (mysql).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "PostgreSQL" (postgresql).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Elastic" (elastic).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Redis" (redis).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Apache Cassandra" (apache_cassandra).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "SQLite" (sqlite).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Neo4j" (neo4j).
[2025-05-20 15:29:07] [INFO] [project.py:123] get_project_list_from_config: Ignoring the project "Djangoproject" (django).
[2025-05-20 15:29:07] [INFO] [project.py:663] scrape_vulnerabilities_from_cve_details: Collecting the vulnerabilities for the "Glibc" project (72, 767):
[2025-05-20 15:29:11] [INFO] [project.py:696] scrape_vulnerabilities_from_cve_details: Scraping hub page 1 of 1...
[2025-05-20 15:29:11] [INFO] [project.py:697] scrape_vulnerabilities_from_cve_details: https://www.cvedetails.com/vulnerability-list.php?vendor_id=72&product_id=767
[2025-05-20 15:29:15] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 1 of 25: "CVE-2025-0395" from "https://www.cvedetails.com/cve/CVE-2025-0395"...
[2025-05-20 15:29:19] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 2 of 25: "CVE-2023-25139" from "https://www.cvedetails.com/cve/CVE-2023-25139"...
[2025-05-20 15:29:23] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 3 of 25: "CVE-2023-6780" from "https://www.cvedetails.com/cve/CVE-2023-6780"...
[2025-05-20 15:29:26] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 4 of 25: "CVE-2023-6779" from "https://www.cvedetails.com/cve/CVE-2023-6779"...
[2025-05-20 15:29:29] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 5 of 25: "CVE-2023-6246" from "https://www.cvedetails.com/cve/CVE-2023-6246"...
[2025-05-20 15:29:30] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 6 of 25: "CVE-2023-5156" from "https://www.cvedetails.com/cve/CVE-2023-5156"...
[2025-05-20 15:29:33] [ERROR] [project.py:290] find_full_git_commit_hash: Failed to find the full version of the commit hash "" with the error: GitCommandError(['git', 'show', '--format=%H', '--no-patch', ''], 128, b"fatal: ambiguous argument '': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'", b'')
[2025-05-20 15:29:33] [ERROR] [cve.py:545] handle_git_urls: --> Could not find a valid commit hash in "https://sourceware.org/git/?p=glibc.git;a=commitdiff;h=ec6b95c3303c700eb89eebeda2d7264cc184a796".
[2025-05-20 15:29:35] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 7 of 25: "CVE-2023-4911" from "https://www.cvedetails.com/cve/CVE-2023-4911"...
[2025-05-20 15:29:36] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 8 of 25: "CVE-2023-4813" from "https://www.cvedetails.com/cve/CVE-2023-4813"...
[2025-05-20 15:29:39] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 9 of 25: "CVE-2023-4806" from "https://www.cvedetails.com/cve/CVE-2023-4806"...
[2025-05-20 15:29:41] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 10 of 25: "CVE-2023-4527" from "https://www.cvedetails.com/cve/CVE-2023-4527"...
[2025-05-20 15:29:43] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 11 of 25: "CVE-2023-0687" from "https://www.cvedetails.com/cve/CVE-2023-0687"...
[2025-05-20 15:29:47] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 12 of 25: "CVE-2022-39046" from "https://www.cvedetails.com/cve/CVE-2022-39046"...
[2025-05-20 15:29:50] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 13 of 25: "CVE-2022-23219" from "https://www.cvedetails.com/cve/CVE-2022-23219"...
[2025-05-20 15:29:55] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 14 of 25: "CVE-2022-23218" from "https://www.cvedetails.com/cve/CVE-2022-23218"...
[2025-05-20 15:30:00] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 15 of 25: "CVE-2021-43396" from "https://www.cvedetails.com/cve/CVE-2021-43396"...
[2025-05-20 15:30:01] [ERROR] [project.py:290] find_full_git_commit_hash: Failed to find the full version of the commit hash "" with the error: GitCommandError(['git', 'show', '--format=%H', '--no-patch', ''], 128, b"fatal: ambiguous argument '': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'", b'')
[2025-05-20 15:30:01] [ERROR] [cve.py:545] handle_git_urls: --> Could not find a valid commit hash in "https://sourceware.org/git/?p=glibc.git%3Ba=commit%3Bh=ff012870b2c02a62598c04daa1e54632e020fd7d".
[2025-05-20 15:30:01] [ERROR] [project.py:290] find_full_git_commit_hash: Failed to find the full version of the commit hash "" with the error: GitCommandError(['git', 'show', '--format=%H', '--no-patch', ''], 128, b"fatal: ambiguous argument '': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'", b'')
[2025-05-20 15:30:01] [ERROR] [cve.py:545] handle_git_urls: --> Could not find a valid commit hash in "https://sourceware.org/git/?p=glibc.git;a=commit;h=ff012870b2c02a62598c04daa1e54632e020fd7d".
[2025-05-20 15:30:02] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 16 of 25: "CVE-2021-38604" from "https://www.cvedetails.com/cve/CVE-2021-38604"...
[2025-05-20 15:30:05] [ERROR] [project.py:290] find_full_git_commit_hash: Failed to find the full version of the commit hash "" with the error: GitCommandError(['git', 'show', '--format=%H', '--no-patch', ''], 128, b"fatal: ambiguous argument '': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'", b'')
[2025-05-20 15:30:05] [ERROR] [cve.py:545] handle_git_urls: --> Could not find a valid commit hash in "https://sourceware.org/git/?p=glibc.git;a=commit;h=b805aebd42364fe696e417808a700fdb9800c9e8".
[2025-05-20 15:30:05] [ERROR] [project.py:290] find_full_git_commit_hash: Failed to find the full version of the commit hash "" with the error: GitCommandError(['git', 'show', '--format=%H', '--no-patch', ''], 128, b"fatal: ambiguous argument '': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'", b'')
[2025-05-20 15:30:05] [ERROR] [cve.py:545] handle_git_urls: --> Could not find a valid commit hash in "https://sourceware.org/git/?p=glibc.git;a=commit;h=4cc79c217744743077bf7a0ec5e0a4318f1e6641".
[2025-05-20 15:30:06] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 17 of 25: "CVE-2021-35942" from "https://www.cvedetails.com/cve/CVE-2021-35942"...
[2025-05-20 15:30:08] [ERROR] [project.py:290] find_full_git_commit_hash: Failed to find the full version of the commit hash "" with the error: GitCommandError(['git', 'show', '--format=%H', '--no-patch', ''], 128, b"fatal: ambiguous argument '': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'", b'')
[2025-05-20 15:30:08] [ERROR] [cve.py:545] handle_git_urls: --> Could not find a valid commit hash in "https://sourceware.org/git/?p=glibc.git;a=commit;h=5adda61f62b77384718b4c0d8336ade8f2b4b35c".
[2025-05-20 15:30:08] [ERROR] [project.py:290] find_full_git_commit_hash: Failed to find the full version of the commit hash "" with the error: GitCommandError(['git', 'show', '--format=%H', '--no-patch', ''], 128, b"fatal: ambiguous argument '': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'", b'')
[2025-05-20 15:30:08] [ERROR] [cve.py:545] handle_git_urls: --> Could not find a valid commit hash in "https://sourceware.org/git/?p=glibc.git%3Ba=commit%3Bh=5adda61f62b77384718b4c0d8336ade8f2b4b35c".
[2025-05-20 15:30:09] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 18 of 25: "CVE-2021-33574" from "https://www.cvedetails.com/cve/CVE-2021-33574"...
[2025-05-20 15:30:14] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 19 of 25: "CVE-2021-27645" from "https://www.cvedetails.com/cve/CVE-2021-27645"...
[2025-05-20 15:30:18] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 20 of 25: "CVE-2021-3999" from "https://www.cvedetails.com/cve/CVE-2021-3999"...
[2025-05-20 15:30:21] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 21 of 25: "CVE-2021-3998" from "https://www.cvedetails.com/cve/CVE-2021-3998"...
[2025-05-20 15:30:25] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 22 of 25: "CVE-2021-3326" from "https://www.cvedetails.com/cve/CVE-2021-3326"...
[2025-05-20 15:30:28] [ERROR] [project.py:290] find_full_git_commit_hash: Failed to find the full version of the commit hash "" with the error: GitCommandError(['git', 'show', '--format=%H', '--no-patch', ''], 128, b"fatal: ambiguous argument '': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'", b'')
[2025-05-20 15:30:28] [ERROR] [cve.py:545] handle_git_urls: --> Could not find a valid commit hash in "https://sourceware.org/git/?p=glibc.git;a=commit;h=7d88c6142c6efc160c0ee5e4f85cde382c072888".
[2025-05-20 15:30:30] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 23 of 25: "CVE-2020-29573" from "https://www.cvedetails.com/cve/CVE-2020-29573"...
[2025-05-20 15:30:33] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 24 of 25: "CVE-2020-29562" from "https://www.cvedetails.com/cve/CVE-2020-29562"...
[2025-05-20 15:30:38] [INFO] [project.py:723] scrape_vulnerabilities_from_cve_details: Scraping the CVE page 25 of 25: "CVE-2020-27618" from "https://www.cvedetails.com/cve/CVE-2020-27618"...
[2025-05-20 15:30:46] [INFO] [collect_vulnerabilities.py:121] main: Finished running for the project "Glibc".
[2025-05-20 15:30:46] [INFO] [collect_vulnerabilities.py:123] main: Finished running.
